daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/InsertOrdMap.scala

```diff
@@ -12,57 +15,37 @@ package com.digitalasset.daml.lf.data
   *  insert: O(1)
   *  remove: O(n)
   */
-import scala.collection.immutable.{HashMap, Map, Queue}
-
-sealed abstract class InsertOrdMap[K, +V] extends Map[K, V] {
-  def _keys: Queue[K]
-  def _hashMap: Map[K, V]
+final class InsertOrdMap[Key, +Value] private (
+    override val keys: Queue[Key],
+    hashMap: HashMap[Key, Value])
+    extends Map[Key, Value] {
```

Comment:
Did you mean to extend `Map` rather than `immutable.Map`?

Reply:
fixed